src/multiplex/pending.rs

```diff
@@ -0,0 +1,69 @@
+use std::{
+    fmt,
+    sync::{atomic, Arc},
+};
+
+use tokio::sync::oneshot;
+
+use crate::wrappers::ClientResponse;
+
+/// A type used to track in-flight requests.
+///
+/// Each pending response contains an associated `Tag` that is provided
+/// by the [`TagStore`], which is used to uniquely identify a request/response pair.
+///
+/// # Drop
+///
+/// On `drop` this type will internally decrement the internal in-flight count and
+/// will return `Error::Cancelled` to the caller.
+pub struct Pending<Response> {
+    in_flight: Arc<atomic::AtomicUsize>,
+    inner: Option<PendingInner<Response>>,
```

Comment:
Why do we need an `Option` ?

Reply:
To support drop otherwise we can't move the sender and impl Drop. `rustc --explain E0509` for more